src/sage/graphs/generic_graph.py

```diff
@@ -1926,7 +1936,10 @@ def adjacency_matrix(self, sparse=None, vertices=None, *, base_ring=None, **kwds
                 sparse = False
 
         if vertices is None:
-            vertices = self.vertices(sort=True)
+            try:
+                vertices = self.vertices(sort=True)
+            except TypeError:
+                raise TypeError ("Vertices are not comparable, you must used the ``vertices`` parameter of the ``adjacency_matrix`` method to specify an ordering")
```

Comment:
a better error message could be
```diff
-                raise TypeError ("Vertices are not comparable, you must used the ``vertices`` parameter of the ``adjacency_matrix`` method to specify an ordering")
+                raise TypeError("Vertex labels are of different types and so "
+                                 "not comparable.  You must specify an ordering "
+                                 "using parameter ``vertices``.")
```

Reply:
I am not sure I agree with this proposition because
- objects can have the same type and not be comparable
- objects can have different types and be comparable
At this point we do not know if the vertices have different types (we just know that they are not comparable), and even if they are it does not imply that they are not comparable.

Reply:
Then just write that `Vertex labels are not comparable. You must specify an ordering using parameter vertices` ?

Reply:
done in b39d8a3